lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
@karies is the expert here. Axel, what would be the right thing to do if we run in non-c++ mode here?